Update built in reducers (#871)
kokkos/kokkos-core-wiki: docs/source/API/core/builtinreducers/MinFirstLoc.rst

``MinFirstLoc``
===============

.. role:: cpp(code)
    :language: cpp

> Specific implementation of `ReducerConcept <ReducerConcept.html>`_ storing the minimum value with an index

Reviewer: ```suggestion
Specific implementation of `ReducerConcept <ReducerConcept.html>`_ storing the minimum value and the first index satisfying a condition.
```

Header File: ``<Kokkos_Core.hpp>``

Usage
-----

.. code-block:: cpp

   MinFirstLoc<T,I,S>::value_type result;
   parallel_reduce(N,Functor,MinFirstLoc<T,I,S>(result));

Synopsis
--------

.. code-block:: cpp

   template<class Scalar, class Index, class Space>
   class MinFirstLoc{
     public:
       using reducer = MinFirstLoc;
       using value_type = ValLocScalar<typename std::remove_cv<Scalar>::type,

>                                typename std::remove_cv<Index>::type >;

Reviewer 2: ```suggestion
                               typename std::remove_cv<Index>::type>;
```

   .. cpp:type:: result_view_type

      A ``Kokkos::View`` referencing the reduction result

   .. rubric:: Constructors

   .. cpp:function:: KOKKOS_INLINE_FUNCTION MinFirstLoc(value_type& value_);

      Constructs a reducer which references a local variable as its result location.

   .. cpp:function:: KOKKOS_INLINE_FUNCTION MinFirstLoc(const result_view_type& value_);

      Constructs a reducer which references a specific view as its result location.

   .. rubric:: Public Member Functions

   .. cpp:function:: KOKKOS_INLINE_FUNCTION void join(value_type& dest, const value_type& src) const;

      Store minimum with the first index of ``src`` and ``dest`` into ``dest``:  ``dest = (src.val < dest.val) ? src :dest;``.

   .. cpp:function:: KOKKOS_INLINE_FUNCTION void init(value_type& val) const;

      Initialize ``val.val`` using the ``Kokkos::reduction_identity<Scalar>::min()`` method. The default implementation sets ``val=<TYPE>_MAX``.
      Initialize ``val.loc`` using the ``Kokkos::reduction_identity<Index>::min()`` method. The default implementation sets ``val=<TYPE>_MAX``.

   .. cpp:function:: KOKKOS_INLINE_FUNCTION value_type& reference() const;

>       Returns a reference to the result provided in class constructor.

Reviewer 2: ```suggestion
      Returns a reference to the result provided in the class constructor.
```

   .. cpp:function:: KOKKOS_INLINE_FUNCTION result_view_type view() const;

>       Returns a view of the result place provided in class constructor.

Reviewer 2: ```suggestion
      Returns a view of the result provided in the class constructor.
```

Additional Information
^^^^^^^^^^^^^^^^^^^^^^

> * ``MinFirstLoc<T,I,S>::value_type`` is Specialization of ValLocScalar on non-const ``T`` and non-const ``I``

Reviewer 2: ```suggestion
```
We already mention that above.
